pkg/cluster/cluster_info_scraper.go

```diff
@@ -326,11 +326,18 @@ func (s *ClusterScraper) GetResourcesPaginated(
 }
 
 func (s *ClusterScraper) GetKubernetesServiceID() (svcID string, err error) {
+	if cachedSvcID, exists := s.cache.Get(kubernetesServiceName); exists {
+		svcID = cachedSvcID.(string)
+		glog.V(4).Infof("Using cached service uid: %s for kubernetes service name: %s", svcID, kubernetesServiceName)
+		return
+	}
 	svc, err := s.CoreV1().Services(k8sDefaultNamespace).Get(context.TODO(), kubernetesServiceName, metav1.GetOptions{})
 	if err != nil {
 		return
 	}
 	svcID = string(svc.UID)
+	glog.V(4).Infof("No cached service uid value found for kubernetes service name: %s", kubernetesServiceName)
+	s.cache.Set(kubernetesServiceName, svcID, 0)
```

Comment:
Using cache ttl as 12 hours, I feel this is good enough. Let me know if any suggestions on this ?

Reply:
I didn't realize that there is a ttl which is not necessary for our use case. But I think this is fine.

Reply:
If you never want to expire the cache, I can set the duration to -1 instead of 0

Reply:
Yes, let's set it to -1 as the clusterID should never change.